server/tasks/stream_inference.py

```diff
@@ -21,16 +23,39 @@
 
 logger = init_logger(__name__)
 
+settings = Settings()
+
 REDIS_URL = os.getenv("REDIS_URL", "redis://localhost:6379/0")
 app = Celery(
     "worker",
     broker=REDIS_URL,
     backend=REDIS_URL,
     accept_content=["json", "pickle"],
     broker_connection_retry_on_startup=True,
+    worker_max_memory_per_child=settings.celery_worker_max_memory_per_child_kb,
```

Comment:
How does per-child work? If I run `celery --concurrency 10` will it allow this much memory per child? 
I believe we run with concurrency 1 on infra side and just have more pod replicas instead, but worth checking

Reply:
We do run with `--concurrency 10` on infra side as well. Unfortunately there's no analogous setting that tracks memory for the whole worker, not the "child worker"/"process worker", so this will not catch a case where eg there are 4 children each holding 0.5GB which will still OOM. But it's not tenable to set this flag at (1.725GB memory limit / 10 children ~= 172MB), because each child uses ~250MB minimum even on startup. I can see a case for lowering it further than 1GB, but I don't want to have to think about this when changing `--concurrency` too.

Reply:
I just feel that since we are running with 1- "children" this wont actually catch anything :(

Reply:
you mean running with >1 children?

Could set the limit to say 300MB, so as soon as the memory starts growing it gets restarted.  I think it's a good bet that children will get different amounts of usage, so one will hit the limit first, it's not like they'll all leak at exactly the same rate.

Reply:
Yep meant "10" not "1-"